app/controllers/answers_controller.rb

```diff
@@ -0,0 +1,27 @@
+class AnswersController < ApplicationController
+  def create
+    question = Question.find(params[:question_id])
+    answer = question.answers.build(answer_params)
+    answer.user_id = current_user.id
+    if answer.save
+      redirect_to question_path(question)
+    else
+      redirect_to new_question_answer_path
+    end
+  end
+
+  def select
+    answer = Answer.find(params[:answer_id])
+    question = answer.question
+    question.update(answered: true)
+    answer.update(selected: true)
```

Comment:
You could be more DRY here - instead of having an answered column on your questions, you could make it a virtual attribute on question using your AR associations. For example on question:
```
def answered?
  answers.where(selected: true).count > 0
end
```

That way you can't end up with inconsistent data if one gets updated but not the other.